### DISCUSSION TITLE
通过Docker部署的机器人如何修改配置？

### DISCUSSION THREAD

**Discussion by QLozin** [nonebot/discussions#287] [问答] [answered]
经历了各种奇怪事件后终于把机器人成功部署到Docker里了，但是如何修改配置呢？（如修改监听端口、增加超级用户）
我注意到容器内部存在`app`文件夹，修改里面的`.env.prod`文件（使用的是这个文件）后重启容器不能实现。修改拉起容器的文件夹宿主机的文件也无济于事。
![app文件夹](https://user-images.githubusercontent.com/53658214/209941389-6482cfd0-69de-498a-b7d3-eca338ca0569.png)
尝试运行容器根目录下的`start-reload.sh`会提示奇怪的错误
![ERROR](https://user-images.githubusercontent.com/53658214/209941672-75f3e8df-1d43-4697-afb2-3f718977b33e.png)
网络上关于机器人Docker化的教程十分稀少，官方文档也只到拉起容器这一步。
是否意味着我必须预先测试好再进行部署，而无法直接修改？
希望得到帮助。

**Comment by yanyongyu** [comment, marked as answer]
请通过docker的env传递配置，如`docker run -e XXX=xxx image`或者docker compose文件中配置env或env_file

建议看下docker和docker compose官方文档

**Reply by QLozin** [reply]
感谢您的回复。
我前去查看了一些用法，还有一些问题没有弄明白。
我注意到，使用`nb-cli`的`create`方法创建的文件自身已经在`docker-compose.yml`文件里有`env_file`并且能正常运作
但我所希望的是在宿主机里配置它，就像挂载卷一样，宿主机对配置文件的修改能够直接同步到容器内部，或者在容器内部对配置文件进行修改也可以（但是我找不到这个文件）
使用`docker run`的方式或`docker-compose`的方式只能在生成容器时进行一次性的配置，此后再次更新好像就要重新生成一个容器了。
感谢您的帮助！

**Reply by yanyongyu** [reply]
修改配置就应该重建容器 `docker compose up -d --build`会自动更新

**Reply by QLozin** [reply]
感谢您的回复。
我已经知晓。
因为之前使用别的程序部署（如`nginx`、`alist`）时可以通过挂载卷的方式来同步配置文件，所以产生这个疑问。
现在该疑问已被解答。
